nixos/lib/eval-config-standard.nix

```diff
@@ -0,0 +1,68 @@
+# DO NOT IMPORT. Use nixpkgsFlake.lib.nixos, or import (nixpkgs + "/nixos/lib")
+{ lib }: # read -^
+
+let
+
+  # Base: Use the miminal module evaluation
+  evalModulesMinimal = (import ./default.nix {
+    inherit lib;
+    # Implicit use of feature is noted in implementation.
+    featureFlags.minimalModules = { };
+  }).evalModules;
+
+  baseModules = import ../modules/module-list.nix;
+
+  /*
+    Invoke NixOS. This is the official entrypoint for evaluating a standard
+    NixOS. All additional configuration or implementations has to be passed
+    via the module system as the `modules` parameter or imported using
+    `imports` in modules.
+
+    An official entrypoint clarifies the public interface of a NixOS
+    evaluation and allows the ecosystem to converge.
+
+    Parameters:
+
+      modules:        A list of modules that constitute the configuration.
+
+      specialArgs:    An attribute set of module arguments. Unlike
+                      `config._module.args`, these are available for use in
+                      `imports`.
+                      `config._module.args` should be preferred when possible.
+
+    Return:
+
+      An attribute set containing `config.system.build.toplevel` among other
+      attributes. See `lib.evalModules` in the Nixpkgs library.
+
+   */
+  evalSystem = args@{ prefix ? [], modules ? [], specialArgs ? {} }: let
+
+    # Locate Modules: Ensure custom modules are located to guarantee a better user experience.
+    #                 Non-located modules are inline modules and ones imported from value instead
+    #                 of a file reference from which the importer could infer the location.
+    #                 For example when idiomatically consuming a public interface via
+    #                 `flake.nixosModules.my-module` instead of a private one via - say -
+    #                 `(flake + /modules/my-module.nix)`.
+    modulesLocation = (builtins.unsafeGetAttrPos "modules" args).file or null;
```

Comment:
The caller could "override" this behavior by pre-applying `setDefaultModuleLocation` to what would otherwise by the `modules`. So not exposing this as a parameter is ok. Perhaps it should be mentioned though? It's a nice thing to do for framework developers.

Reply:
comment addressed.